File: src/main/java/gargoyle/netsquares/util/log/JULLogger.java
package gargoyle.netsquares.util.log;

final class JULLogger implements Logger {
    private final java.util.logging.Logger logger;

    public JULLogger(String name) {
        logger = java.util.logging.Logger.getLogger(name);
    }

    @Override
    public void log(Level level, String message, Throwable exception) {
        logger.log(toLevel(level), message, exception);
    }

    private java.util.logging.Level toLevel(Level level) {
        switch (level) {
            case ERROR: {
                return java.util.logging.Level.SEVERE;
            }
            case WARN: {
                return java.util.logging.Level.WARNING;
            }
            case INFO: {
                return java.util.logging.Level.INFO;
            }
            case DEBUG: {
                return java.util.logging.Level.FINE;
            }
            case TRACE: {
                return java.util.logging.Level.FINER;
            }
        }
        throw new RuntimeException(String.valueOf(level));
    }
}

File: src/main/kotlin/gargoyle/netsquares/util/log/JULLogger.kt
package gargoyle.netsquares.util.log

internal class JULLogger(name: String) : Logger {
    private val logger: java.util.logging.Logger = java.util.logging.Logger.getLogger(name)

    override fun log(level: Level, message: String, exception: Throwable?) =
        logger.log(toLevel(level), message, exception)

    private fun toLevel(level: Level): java.util.logging.Level =
        when (level) {
            Level.ERROR -> java.util.logging.Level.SEVERE
            Level.WARN -> java.util.logging.Level.WARNING
            Level.INFO -> java.util.logging.Level.INFO
            Level.DEBUG -> java.util.logging.Level.FINE
            Level.TRACE -> java.util.logging.Level.FINER
        }
}
